src/middlewares/permissions.js

```diff
@@ -0,0 +1,44 @@
+import Joi from '@hapi/joi';
+import { resource } from '../models';
+import { permissionType, checkPermission, setPermissionSchema } from '../validation/permissions';
+
+export const permit = async (req, res, next) => {
```

Comment:
Can you explain the permisson concept and logic?

Reply:
Yes I can. I have created a google doc for the team on it already.